src/cli/cli.cpp

```diff
@@ -2621,14 +2639,33 @@ otError Interpreter::ProcessMasterKey(uint8_t aArgsLength, Arg aArgs[])
 
     if (aArgsLength == 0)
     {
-        OutputBytes(otThreadGetMasterKey(mInstance)->m8);
+        const otMasterKey *masterKey = otThreadGetMasterKey(mInstance);
+
+        if(otPlatCryptoGetType() == OT_CRYPTO_TYPE_PSA)
+        {
+            uint8_t mMasterKeyBuffer[OT_MASTER_KEY_SIZE];
+            size_t mKeyLen;
+
+            Error err = otPlatCryptoExportKey(masterKey->mKeyMaterial.m32,
+                                                mMasterKeyBuffer,
+                                                OT_MASTER_KEY_SIZE,
+                                                &mKeyLen); 
+
+            (void)err;
+            OutputBytes(mMasterKeyBuffer);  
```

Comment:
I think memset of a stack buffer will most likely be optimized out: https://godbolt.org/z/zMxzWMn3G. We probably should use `memset_s` or an alternative available on a given platform. As it may not be straightforward, maybe it should be done in a separate PR?